packages/plugins/FileService/src/SNSAdaptor/components/FileList.tsx

```diff
@@ -0,0 +1,208 @@
+                    })
+                }
+            }
+        },
+        [showConfirm, t],
+    )
+    const showPrompt = useShowRenameDialog()
+    const handleRename = useCallback(
+        async (file: FileInfo) => {
+            const newName = await showPrompt({
+                currentName: file.name,
+                message: t.rename_validation(),
+            })
+            if (newName) {
+                await PluginFileServiceRPC.renameFile(file.id, newName)
+                refetchFiles()
+            }
+        },
+        [showPrompt, t],
+    )
+
+    return (
+        <section className={cx(classes.container, className)} {...rest}>
+            <List className={classes.list} classes={{ root: classes.listRoot }}>
+                {files.map((file) => (
+                    <ListItem key={file.id} className={classes.listItem} classes={{ root: classes.itemRoot }}>
+                        {uploadStateMap[file.id] ? (
+                            <UploadingFile
+                                className={classes.file}
+                                file={file}
+                                progress={uploadStateMap[file.id]?.progress ?? 0}
+                            />
+                        ) : (
+                            <ManageableFile
+                                className={classes.file}
+                                file={file}
+                                onDelete={handleDelete}
+                                onRename={handleRename}
+                                onDownload={onDownload}
+                                onSend={onSend}
+                            />
+                        )}
+                    </ListItem>
+                ))}
+            </List>
+        </section>
+    )
+}
+
+interface SelectableFileListProps extends Omit<FileListBaseProps, 'onChange' | 'selected'> {
+    selectedIds?: string[]
+    onChange?(selectedIds: string[]): void
+}
+
+const FILE_LIMIT = 5
+export const SelectableFileList: FC<SelectableFileListProps> = ({
+    files,
+    className,
+    selectedIds = EMPTY_LIST,
+    onChange,
+    ...rest
+}) => {
+    const { classes, cx } = useStyles()
+
+    const selectedIdsRef = useRef(selectedIds)
+    const filesRef = useRef(files)
+    const onChangeRef = useRef(onChange)
+
+    selectedIdsRef.current = selectedIds
+    filesRef.current = files
+    onChangeRef.current = onChange
+
+    const handleChange = useCallback((fileId: string, checked: boolean) => {
+        const oldIds = selectedIdsRef.current
+        const newIds = filesRef.current.map((x) => x.id).filter((id) => (id === fileId ? checked : oldIds.includes(id)))
+        onChangeRef.current?.(newIds)
+    }, [])
+
+    return (
+        <section className={cx(classes.container, className)} {...rest}>
+            <List className={classes.list} classes={{ root: classes.listRoot }}>
+                {files.map((file) => (
```

Comment:
What if `files` is an empty array?

Reply:
Suppose been checked at upper level component.